src/utils/subgraphQuery.ts

```diff
@@ -83,3 +83,41 @@ export const getValidContact = async (
     );
   }
 };
+
+export const checkProtectedDataValidity = async (
+  graphQLClient: GraphQLClient,
+  protectedData: string
+): Promise<boolean> => {
+  try {
+    const SchemaFilteredProtectedData = gql`
+      query ($requiredSchema: [String!]!, $id: String!) {
+        protectedDatas(
+          where: {
+            transactionHash_not: "0x"
+            schema_contains: $requiredSchema
+            id: $id
+          }
+        ) {
+          id
+        }
+      }
+    `;
+
+    const variables = {
+      requiredSchema: ['email:string'],
+      id: protectedData,
+    };
+
+    const protectedDataResultQuery: GraphQLResponse =
+      await graphQLClient.request(SchemaFilteredProtectedData, variables);
```

Comment:
it's better to refactor the code into seperate functions or constants, that way we have a readable and modular code
```suggestion
const getQuery = (requiredSchema: string[], id: string) => {
  return gql`
    query CheckProtectedDataValidity($requiredSchema: [String!]!, $id: String!) {
      protectedDatas(
        where: {
          transactionHash_not: "0x",
          schema_contains: $requiredSchema,
          id: $id
        }
      ) {
        id
      }
    }
  `;
};

const createVariables = (protectedData: string) => {
  return {
    requiredSchema: ['email:string'],
    id: protectedData,
  };
};
export const checkProtectedDataValidity = async (
  graphQLClient: GraphQLClient,
  protectedData: string
): Promise<boolean> => {
  try {
    const query = getQuery(['email:string'], protectedData);
    const variables = createVariables(protectedData);

    const result: GraphQLResponse = await graphQLClient.request(query, variables);
```

Reply:
Commit: 95f8780d1bf1c6dc257a9c2e8de19d13bd71ceef